From 2e88c6a805fc5311e27e0f6efe243842634052ab Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Sun, 9 Jan 2022 09:13:37 +0100
Subject: [PATCH] ALSA: hda: Fix dependencies of CS35L41 on SPI/I2C buses
Git-commit: 2e88c6a805fc5311e27e0f6efe243842634052ab
Patch-mainline: v5.17-rc1
References: bsc#1203699

CS35L41 SPI and I2C drivers depend on those buses, hence they have to
have dependencies in Kconfig; otherwise it may result in missing
symbols.

Fixes: 7b2f3eb492da ("ALSA: hda: cs35l41: Add support for CS35L41 in HDA systems")
Reported-by: kernel test robot <lkp@intel.com>
Link: https://lore.kernel.org/r/20220109081337.30623-1-tiwai@suse.de
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 sound/pci/hda/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/pci/hda/Kconfig b/sound/pci/hda/Kconfig
index 68effb74866c..febe1c2b7d9a 100644
--- a/sound/pci/hda/Kconfig
+++ b/sound/pci/hda/Kconfig
@@ -96,6 +96,7 @@ config SND_HDA_SCODEC_CS35L41
 
 config SND_HDA_SCODEC_CS35L41_I2C
 	tristate "Build CS35L41 HD-audio side codec support for I2C Bus"
+	depends on I2C
 	depends on ACPI
 	depends on SND_SOC
 	select SND_HDA_GENERIC
@@ -110,6 +111,7 @@ comment "Set to Y if you want auto-loading the side codec driver"
 
 config SND_HDA_SCODEC_CS35L41_SPI
 	tristate "Build CS35L41 HD-audio codec support for SPI Bus"
+	depends on SPI_MASTER
 	depends on ACPI
 	depends on SND_SOC
 	select SND_HDA_GENERIC
-- 
2.35.3

